app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -1113,11 +1113,17 @@ private void sendMessage() {
                 return;
             } else {
                 try {
-                    Cursor streamCursor = makeStreamCursor(streamActv.getText().toString());
-                    if (streamCursor.getCount() == 0) {
+                    //check whether stream exists or not
+                    if (!isStreamExist(streamActv.getText().toString())) {
```

Comment:
Oh, I missed that... But I think doing 2 database calls before sending message is expensive. @kunall17 How about we move `isStreamExist` to https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/activities/ZulipActivity.java#L1168 ?